source/s3_meta_request.c

```diff
@@ -236,9 +236,13 @@ static void s_s3_request_on_finish(
 
     if (request_binding->input_body) {
         /* close the input file stream now, request has finished, we will not read from there anymore */
-        aws_input_stream_destroy(request_binding->input_body);
+        aws_input_stream_release(request_binding->input_body);
         request_binding->input_body = NULL;
     }
+    if (request_binding->copied_message) {
+        aws_http_message_release(request_binding->copied_message);
+        request_binding->copied_message = NULL;
+    }
```

Comment:
two trivial things:

1) was this a memory leak you caught? I don't see similar code in the before-view
2) these 4 lines can just be `request_binding->copied_message = aws_http_message_release(request_binding_copied_message);`

Reply:
yes, it was uncovered by the input stream change